src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -98,29 +106,106 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
                     LogMetricsFromAuthResult(authenticationResult, AuthenticationRequestParameters.RequestContext.Logger);
                     LogSuccessfulTelemetryToClient(authenticationResult, telemetryEventDetails, telemetryClients);
 
+                    OpenTelemetryClient.LogSuccessMetrics(
+                        ServiceBundle.PlatformProxy.GetProductName(), 
+                        AuthenticationRequestParameters.AppConfig.ClientId, 
+                        authenticationResult.AuthenticationResultMetadata, 
+                        GetCacheLevel(authenticationResult).ToString());
+
+                    LogMsalSuccessTelemetryToOTel(authenticationResult);
+
                     return authenticationResult;
                 }
                 catch (MsalException ex)
                 {
                     apiEvent.ApiErrorCode = ex.ErrorCode;
                     AuthenticationRequestParameters.RequestContext.Logger.ErrorPii(ex);
                     LogMsalErrorTelemetryToClient(ex, telemetryEventDetails, telemetryClients);
+
+                    OpenTelemetryClient.LogToFailureCounter(
+                        ServiceBundle.PlatformProxy.GetProductName(), 
+                        AuthenticationRequestParameters.AppConfig.ClientId, 
+                        ex.ErrorCode);
```

Comment:
What will it be in the case of an MsalClientException? can there be such an exception?

Reply:
It should be like this; 

- MsalServiceException -> the suberror code (e.g. 500015) or if not present the error code
- MsalClientException -> the error code (e.g. invalid_certificate)
- other exceptions -> the type of the exception (e.g. SocketTimeout)

Reply:
All the MsalExceptions have error code so the behavior will be as mentioned above by Bogdan.